geometry.ts

```diff
@@ -81,6 +81,82 @@ export class Segment {
 	}
 }
 
+export class Arc {
+	constructor(
+		public readonly center: Position,
+		public readonly end1: Position,
+		public readonly end2: Position,
+	) { }
+
+	static angleDistance(a1: number, a2: number): number {
+		a1 %= Math.PI * 2;
+		a2 %= Math.PI * 2;
+		if (a2 < a1) {
+			a2 += Math.PI * 2;
+		}
+		return Math.min(a2 - a1, Math.abs(a1 + Math.PI * 2 - a2));
+	}
+
+
+	nearestToArc(
+		q: Position
+	): { t: number, position: Position } {
+		const a1 = Math.atan2(this.end1.y - this.center.y, this.end1.x - this.center.x);
+		let a2 = Math.atan2(this.end2.y - this.center.y, this.end2.x - this.center.x);
+		while (a2 < a1) {
```

Comment:
You can use `if` instead

Reply:
Yeah, I made this mistake in a few places. There are now helpers which generally encapsulate this logic.